fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1521,7 +1560,12 @@ public void inferDistribution(DistributionInfo info) throws DdlException {
         if (info.getBucketNum() == 0) {
             int inferredBucketNum = 0;
             for (BaseTableInfo base : getBaseTableInfos()) {
-                Table table = MvUtils.getTableChecked(base);
+                Optional<Table> tableOptional = MvUtils.getTable(base);
+                if (tableOptional.isEmpty()) {
+                    LOG.warn("base table: {} dropped", base.getReadableString());
```

Comment:
is it a error situation? why not throw a RuntimeException?

Reply:
it is not an error situation. just a protection.